docs/devguide/modules-dev-guide.asciidoc

```diff
@@ -336,6 +354,88 @@ Note that you should follow the convention of naming of fields prefixed with the
 module and fileset name: `{module}.{fileset}.field`, e.g.
 `nginx.access.remote_ip`. Also, please review our <<event-conventions>>.
 
+In 6.6 and later, ingest pipelines can use the `pipeline` processor to delegate
+part of the processing to another sub-pipeline.
+
+[source,json]
+----
+{
+  "pipeline": {
+    "name": "name-of-sub-pipeline"
+  }
+}
+----
+
+This can be useful if you want a fileset to ingest the same _logical_ information
+presented in different formats, e.g. csv vs. json versions of the same log files.
+Imagine a root ingest pipeline that detects the format of a log entry and then conditionally
+delegates further processing of that log entry, depending on the format, to a specific
+sub-pipeline.
+
+[source,json]
+----
+{
+  "processors": [
+    {
+      "grok": {
+        "field": "message",
+        "patterns": [
+          "^%{CHAR:first_char}"
+        ],
+        "pattern_definitions": {
+          "CHAR": "."
+        }
+      }
+    },
+    {
+      "pipeline": {
+        "if": "ctx.first_char == '{'"
+        "name": "json-log-processing-pipeline"
+      }
+    },
+    {
+      "pipeline": {
+        "if": "ctx.first_char != '{'"
+        "name": "plain-log-processing-pipeline"
+      }
+    }
+  ]
+}
+----
+
+In order for the above pipeline to work, Filebeat must load the root pipeline
+as well as any sub-pipelines into Elasticsearch. You can tell Filebeat to do
+so by specifying all the necessary pipelines for the fileset in its `manifest.yml`
+file. The first pipeline in the list is considered to be the root pipeline.
+
+[source,yaml]
+----
+ingest_pipeline:
+  - ingest/main.json
+  - ingest/plain_logs.json
+  - ingest/json_logs.json
+----
+
+Additionally, any `pipeline` processors referencing one of the pipelines in the
```

Comment:
Agreed that the conflicting examples cause confusion. Technically both examples are valid, but the one below, i.e. the one using the `< IngestPipeline "plain_logs" >` syntax, is the one we'd want fileset authors to use when trying to use this new feature (multiple pipelines with an entry point pipeline for a fileset).

The reason I split up the two examples was to first just get the reader introduced to the idea of a pipeline that delegates part of the processing to other pipelines and then explain the idea of the template function. I was wary of introducing **two** new ideas in the same example snippet in case.

But I see now the confusion this can cause, so I agree that we should always show the use of the `< IngestPipeline "plain_logs" >` syntax in the example(s). Perhaps there's some way we could smoosh the examples and explanation of these two ideas together without potentially overwhelming the reader?

Reply:
Perfect, callouts make a lot of sense here, thanks!